### DISCUSSION TITLE
How to make meow-query-replace and meow-query-replace-regexp use the selection at point (or match) as their argument

### DISCUSSION THREAD

**Discussion by Kenneth-Mata** [meow-edit/meow#502] [Q&A] [answered]
Let's say i want to change all instances of use-package using meow, so i press `meow-mark-symbol`, which highlights all instances of "use-package" across the buffer, then i do `meow-query-replace` but now i have to type "meow-mark-symbol" if that's what i want to change, furthermore, it will only replace that single instance if i don't have the whole buffer selected

**Comment by DogLooksGood** [comment]
Any example to demonstrate the way you want it to be?

**Comment by Kenneth-Mata** [comment]
https://github.com/meow-edit/meow/assets/92641443/6bdb981d-0153-40da-8238-b7b496962ba7

I hope this explains what i mean, with isearch i can call a query-replace on the search subject and travel through matches either applying the change or not, with meow-mark (or meow-visit) when i call meow-query-replace it doesn't pick up the subject and even if i say, yank it, it doesn't travel to the others

apparently the particular command is called isearch-query-replace

**Reply by DogLooksGood** [reply, marked as answer]
I see.

You can select with `meow-mark-symbol`, update it with `meow-change`, typing, `ESC`. 
When you exit from the insert state, the whole symbol should be marked automatically.

Then you do `meow-save`, `meow-search`, `meow-replace`, then repeat search and replace.

**Reply by Kenneth-Mata** [reply]
That works perfectly, and it wraps around the file too, thank you.

**Comment by Esnos33** [comment]
You can also `mark-page`, `meow-visit` on `use-package`, `meow-change`, type and `ESC`. This is helix workflow.

**Reply by LovingSalamander** [reply]
After `meow-visit`, this is highlighting all instances of `use-package` for me, with an indicator like [1/N] next to the current selection. But when I `meow-change` and hit ESC, it only changes the selection, not all highlighted instances.

**Reply by DogLooksGood** [reply]
To do multi-edit, you need a secondary selection first. `meow-mark-{word/symbol}` and `meow-search` can be used together for searching, that's why there's a highlight and indicator.
